File: internal/indexer/testdata/go/services/user_service.go
package services

const MaxWorkers = 4

func ProcessUserData(userID int, data map[string]string) bool {
	if len(data) == 0 {
		return false
	}
	return true
}

func ValidateEmail(email string) bool {
	for _, ch := range email {
		if ch == '@' {
			return true
		}
	}
	return false
}

func FormatUserName(first, last string) string {
	return first + " " + last
}

type UserRepository struct{}

func (r *UserRepository) FindByID(userID int) map[string]string {
	return map[string]string{"id": "1"}
}

func (r *UserRepository) Save(user map[string]string) bool {
	return true
}

File: internal/indexer/testdata/java/services/UserService.java
package services;

import java.util.Map;
import java.util.Optional;

public class UserService {
    public static final int MAX_WORKERS = 4;

    public static boolean processUserData(int userId, Map<String, String> data) {
        return data != null && !data.isEmpty();
    }

    public static boolean validateEmail(String email) {
        return email != null && email.contains("@");
    }

    public static String formatUserName(String first, String last) {
        return first.trim() + " " + last.trim();
    }

    public Optional<Map<String, String>> findById(int userId) {
        return Optional.of(Map.of("id", String.valueOf(userId)));
    }

    public boolean save(Map<String, String> user) {
        return true;
    }
}
